src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+        /// </summary>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="entityName">Current entity's name.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="objectFieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="columnsDerivedFromSourceEntity">Set of columns in this entity whose values should be
+        /// derived from insertion in the source entity (i.e. source entity would have been the referenced entity).</param>
+        /// <param name="columnsToDerive">Set of columns in this entity whose values are to be
+        /// derived from insertion in the this entity or its subsequent referenced entities and returned to the source entity
+        /// so as to provide values for the corresponding referencing fields (i.e. source entity would have been the referencing entity).</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request.</param>
+        /// <param name="parentEntityName">Parent entity's name.</param>
+        /// <param name="sqlMetadataProviderFactory">Metadata provider factory.</param>
+        private static void ValidateObjectFieldNodes(
+            InputObjectType schemaObject,
+            string entityName,
+            IMiddlewareContext context,
+            IReadOnlyList<ObjectFieldNode> objectFieldNodes,
+            RuntimeConfig runtimeConfig,
+            HashSet<string> columnsDerivedFromSourceEntity,
+            HashSet<string> columnsToDerive,
+            int nestingLevel,
+            string parentEntityName,
+            IMetadataProviderFactory sqlMetadataProviderFactory)
+        {
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
+            ISqlMetadataProvider metadataProvider = sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            SourceDefinition sourceDefinition = metadataProvider.GetSourceDefinition(entityName);
+            Dictionary<string, IValueNode?> backingColumnData = MultipleCreateOrderHelper.GetBackingColumnDataFromFields(context, entityName, objectFieldNodes, metadataProvider);
+
+            // Set of columns in the current entity whose values can be derived via:
+            // a. User input
+            // b. Insertion in the source referenced entity (if current entity is a referencing entity for a relationship with its parent entity)
+            // c. Insertion in the target referenced entity (if the current entity is a referencing entity for a relationship with its target entity)
+            //
+            // derivableColumnsFromRequestBody is initialized with the set of columns whose value is specified in the input (a).
+            Dictionary<string, string> derivableColumnsFromRequestBody = new();
```

Comment:
```suggestion
            Dictionary<string, string> currentEntityFieldsPresentInTheRequestBody = new();
```

Reply:
I think the current variable name is apt to represent what it intends to store. Please refer to the comments preceding this variable explaining what all it can store.